src/daggy.js

```diff
@@ -46,22 +46,22 @@
     var typeRep = proto.constructor = {
       toString: typeRepToString,
       prototype: proto,
-      is: isType,
       '@@type': typeName
     };
+    typeRep.is = isType(typeName);
     Object.keys(constructors).forEach(function(tag) {
       var fields = constructors[tag];
       var tagProto = Object.create(proto);
       defProp(tagProto, TAG, tag);
       if (fields.length === 0) {
         typeRep[tag] = makeValue(fields, tagProto, [], 0);
-        typeRep[tag].is = sum$isUnit;
+        typeRep[tag].is = sum$isUnit(typeRep[tag]);
         return;
       }
       typeRep[tag] = makeConstructor(fields, tagProto);
-      typeRep[tag].is = sum$isVariant;
       typeRep[tag][TAG] = tag;
       typeRep[tag][RET_TYPE] = typeName;
+      typeRep[tag].is = sum$isVariant(typeRep[tag]);
```

Comment:
Is there a reason for moving this property assignment?

Reply:
I think I was thinking something like having the `TAG` and `RET_TYPE` props defined before making the checker, but the order doesn't really matter.  I can put it back.